### PR TITLE
Calculate slot and manually set timestamp

### PR DESCRIPTION
**Pull request by meltedblocks**
@lowprivuser could you do CR?

### CODE REVIEW

File: src/extensions/program_test_context.rs [hunk 1]
```diff
@@ -1,28 +1,41 @@
...
+            return Err(ProgramTestError::InvalidWarpSlot);
         }
 
-        let clock: Clock = self.banks_client.get_sysvar().await.unwrap();
-        now = clock.unix_timestamp;
+        let ns_per_slot = self.genesis_config().ns_per_slot();
+        let timestamp_diff_ns = timestamp
+            .checked_sub(now) //calculate time diff
+            .expect("Problem with timestamp diff calculation.")
+            .checked_mul(1000000000) //convert from s to ns
```
Comment (lowprivuser) [inline src/extensions/program_test_context.rs]: would be nice to have a constant here

**Review by lowprivuser — CHANGES_REQUESTED** [review]
looks good, there's just one last little thing left to do

File: src/extensions/program_test_context.rs (hunk 1 above)
Comment (meltedblocks) [reply src/extensions/program_test_context.rs]: done

**Review by lowprivuser — APPROVED** [review]
Looks good, thanks!

**Pull request merged**